### PR TITLE
Apply Prettier

### PR DESCRIPTION
**Pull request by phryneas**
This PR "inverses" our `.prettierignore` file, upgrades `prettier` to `v3`, moves the `.prettierrc` out of the `package.json`, applies the change to `'` vs `"` that we decided upon and then formats the following folders:
* `config` - I hope this one is okay even if we didn't talk about that, the files in there are just really hard to handle without proper formatting
* `integration-tests` (should have been formatted all along)
* `src/react` folders.
* a bunch of config files that were previously unformatted

Probably relevant for review: [diff on the output files](https://github.com/apollographql/apollo-client/actions/runs/5714535541?pr=11111)

I'll add the resulting commit *after we squash merge* to https://github.com/apollographql/apollo-client/blob/main/.git-blame-ignore-revs - adding anything now wouldn't have any impact.

### Checklist:

- [ ] If this PR contains changes to the library itself (not necessary for e.g. docs updates), please include a changeset (see [CONTRIBUTING.md](https://github.com/apollographql/apollo-client/blob/main/CONTRIBUTING.md#changesets))
- [ ] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
- [ ] Make sure all of the significant new logic is covered by tests

### CODE REVIEW

File: ignoreds.mjs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+.concat(readdirSync('./integration-tests', { recursive: true }).map(f => `/integration-tests/${f}`))
+.filter(f => f.includes(".") && !(/node_modules|.yalc|.next|dist/.test(f)))
+// console.log({ignored, files})
+const notIgnored = micromatch(files, '**/*.*', { ignore: ignored})
+
+const ignoredMatches = files.filter(f => !notIgnored.includes(f))
+for (const i of ignoredMatches) {
+  console.log(i)
+}
+
```
Comment (phryneas) [inline ignoreds.mjs:17]: This will be deleted once I'm done, just throwing it in to document what I did here

**Comment by phryneas** [comment]
Probably relevant for review: [diff on the output files](https://github.com/apollographql/apollo-client/actions/runs/5714535541?pr=11111)

**Review by jerelmiller — APPROVED** [review]
🎉

**Review by alessbell — APPROVED** [review]
🥳

**Pull request merged**